storage/backend/ceph/internals.go

```diff
@@ -1,26 +1,69 @@
 package ceph
 
 import (
+	"encoding/json"
 	"fmt"
 	"os/exec"
 	"regexp"
 	"strings"
 
-	"encoding/json"
-
 	"github.com/contiv/volplugin/storage"
 
 	log "github.com/Sirupsen/logrus"
 )
 
+func (c *Driver) lockImage(do storage.DriverOptions) error {
+	poolName := do.Volume.Params["pool"]
+
+	if err := exec.Command("rbd", "lock", "add", do.Volume.Name, do.Volume.Name, "--pool", poolName).Run(); err != nil {
+		log.Debugf("Could not acquire lock for %q: %v", do.Volume.Name, err)
+		return err
+	}
+
+	return nil
+}
+
+func (c *Driver) unlockImage(do storage.DriverOptions) error {
+	poolName := do.Volume.Params["pool"]
+
+	output, err := exec.Command("rbd", "lock", "--format", "json", "list", do.Volume.Name, "--pool", poolName).Output()
```

Comment:
similar to above, using `CombinedOutput()` might be better as that will log stderr as well. And may be we should also log the output in case of failures.